Application crashes if http entity is null

[Urllib.java#L150](https://github.com/liato/android-bankdroid/blob/ef0f91244bdf47e1f2ba0aa2280d12f7fb63c943/bankdroid-legacy/src/main/java/eu/nullbyte/android/urllib/Urllib.java#L150) will throw an `IllegalArgumentException` if the returned `HttpEntity` is null.

To minimize the impact of the fix it is suggested that the method should do a `null` check and return an empty `String` when the `HttpEntity` is `null`.

```java
java.lang.RuntimeException: An error occured while executing doInBackground()
       at android.os.AsyncTask$3.done(AsyncTask.java:300)
       at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
       at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
       at java.util.concurrent.FutureTask.run(FutureTask.java:242)
       at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:841)
Caused by: java.lang.IllegalArgumentException: HTTP entity may not be null
       at org.apache.http.util.EntityUtils.toString(EntityUtils.java:110)
       at org.apache.http.util.EntityUtils.toString(EntityUtils.java:146)
       at eu.nullbyte.android.urllib.Urllib.open(Urllib.java:150)
       at eu.nullbyte.android.urllib.Urllib.open(Urllib.java:147)
       at eu.nullbyte.android.urllib.Urllib.open(Urllib.java:139)
       at com.liato.bankdroid.banking.banks.nordea.Nordea.preLogin(Nordea.java:197)
       at com.liato.bankdroid.banking.banks.nordea.Nordea.login(Nordea.java:239)
       at com.liato.bankdroid.banking.banks.nordea.Nordea.update(Nordea.java:264)
       at com.liato.bankdroid.DataRetrieverTask.doInBackground(DataRetrieverTask.java:89)
       at com.liato.bankdroid.DataRetrieverTask.doInBackground(DataRetrieverTask.java:40)
       at android.os.AsyncTask$2.call(AsyncTask.java:288)
       at java.util.concurrent.FutureTask.run(FutureTask.java:237)
       at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
       at java.lang.Thread.run(Thread.java:841)
```

--- a/bankdroid-legacy/src/main/java/eu/nullbyte/android/urllib/Urllib.java
+++ b/bankdroid-legacy/src/main/java/eu/nullbyte/android/urllib/Urllib.java
@@ -147,7 +147,11 @@ public class Urllib {
         return open(url, postData, false);
     }
     public String open(String url, List<NameValuePair> postData, boolean forcePost) throws ClientProtocolException, IOException {
-        return EntityUtils.toString(openAsHttpResponse(url, postData, forcePost).getEntity());
+        HttpEntity entity = openAsHttpResponse(url, postData, forcePost).getEntity();
+        if(entity == null) {
+            return "";
+        }
+        return EntityUtils.toString(entity);
     }
 
     public HttpResponse openAsHttpResponse(String url, List<NameValuePair> postData, boolean forcePost) throws ClientProtocolException, IOException {

